bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -21,434 +20,288 @@ class Neuron(CMakePackage):
+    # common build dependencies
+    depends_on("bison@3:", type="build")
     depends_on("flex@2.6:", type="build")
+    depends_on("ninja", type="build")
 
-    # Readline became incompatible with Mac so we use neuron internal readline.
-    # HOWEVER, with the internal version there is a bug which makes
-    # Vector.as_numpy() not work!
-    depends_on("readline", when=sys.platform != "darwin")
-
-    # Transient dependency
+    # build and link dependencies
     depends_on("gettext")
-
     depends_on("mpi", when="+mpi")
-    depends_on("py-mpi4py", when="+mpi+python+tests")
     depends_on("ncurses")
-    depends_on("python@2.6:", when="+python", type=("build", "link", "run"))
+    depends_on("readline")
+
+    # dependencies from python variants
+    depends_on("python", when="+python")
     depends_on("py-pytest", when="+python+tests")
-    # Numpy is required for Vector.as_numpy()
-    depends_on("py-numpy", when="+python", type=("build", "run"))
+    depends_on("py-mpi4py", when="+mpi+python+tests")
+    depends_on("py-numpy", when="+python")
     depends_on("py-cython", when="+rx3d", type="build")
-    depends_on("py-pytest-cov", when="+tests@8:")
-    # TODO: newer spack doesn't propogate python package as dependency
-    # (as PYTHONPATH) during build time
-    depends_on("py-sympy@1.3:", type=("build", "run"))
-    # Next two needed in versions containing neuronsimulator/nrn#2235.
+    depends_on("py-pytest-cov", when="+tests")
+    # next two needed after neuronsimulator/nrn#2235.
     depends_on("py-pip", type=("build"), when="@9:")
     depends_on("py-packaging", type=("run"), when="@9:")
 
-    # dependency on coreneuron via submodule
-    depends_on("coreneuron+legacy-unit~caliper", when="@:8.99+coreneuron+legacy-unit~caliper")
-    depends_on("coreneuron~legacy-unit~caliper", when="@:8.99+coreneuron~legacy-unit~caliper")
-    depends_on("coreneuron+legacy-unit+caliper", when="@:8.99+coreneuron+legacy-unit+caliper")
-    depends_on("coreneuron~legacy-unit+caliper", when="@:8.99+coreneuron~legacy-unit+caliper")
-    conflicts("coreneuron", when="@9:")
-
-    # dependencies from coreneuron package
-    depends_on("python", type=("build", "run"))
-    depends_on("boost", when="@8.99:+tests+coreneuron")
-    depends_on("cuda", when="@8.99:+gpu")
-    depends_on("libsonata-report", when="@8.99:+report+coreneuron")
-
-    # for coreneuron: some basic conflicts
-    gpu_compiler_message = "neuron: for gpu build use %pgi or %nvhpc"
+    # dependencies from coreneuron
+    depends_on("boost", when="+coreneuron+tests")
+    depends_on("cuda", when="+coreneuron+gpu")
+    depends_on("py-sympy@1.3:", when="+coreneuron")
+
+    # dependencies for performance measurement
+    depends_on("caliper", when="+caliper")
+
+    # some common conflicts that use can build
+    conflicts("+rx3d", when="~python")
+    conflicts("+gpu", when="~coreneuron")
+    conflicts("+openmp", when="~coreneuron")
```

Comment:
These seem moot with conditional variants.

Reply:
that's being me not up to date! will do!